llama_index/query_engine/graph_query_engine.py

```diff
@@ -27,13 +30,15 @@ def __init__(
         graph: ComposableGraph,
         custom_query_engines: Optional[Dict[str, BaseQueryEngine]] = None,
         recursive: bool = True,
+        similarity_top_k: int = DEFAULT_SIMILARITY_TOP_K,
```

Comment:
tbh, should probably just accept a general dictionary of kwargs, rather than a specific attribute. Then it can be used for any sub-index

Reply:
Happy to change it to kwargs. I thought this way it would be less "permissive" as it only adds 1 more optional arguments, than allowing anything to be passed in. In the implementation, I check and pass it to the underlying index only if it is a Vector, so that it doesn't impact any other use cases. With being explicit it would be easier to understand and less chance of causing errors.

Reply:
Being generic kwargs means we can handle any option for any sub index.

I think this is fine, `as_query_engine(...)` already works fine with any kwargs, even if they are unused.

The other option is setting up the custom query engines mapping instead